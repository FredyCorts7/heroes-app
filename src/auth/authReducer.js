import { types } from '../types/types';

export const authReducer = (state = {}, action) => {
  if (!action) return state;

  switch (action.type) {
    case types.LOGIN:
      return {
        ...action.payload,
        authenticated: true,
      };
    case types.LOGOUT:
      return {
        authenticated: false,
      };
    default:
      return state;
  }
};
