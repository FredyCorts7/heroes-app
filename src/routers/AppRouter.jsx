import React, { useContext } from 'react';
import { BrowserRouter as Router, Switch } from 'react-router-dom';
import { AuthContext } from '../auth/AuthContext';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {
  const {
    user: { authenticated },
  } = useContext(AuthContext);

  return (
    <Router>
      <div>
        <Switch>
          <PublicRoute
            isAuthenticated={authenticated}
            path='/login'
            component={LoginScreen}
          />

          <PrivateRoute
            isAuthenticated={authenticated}
            path='/'
            component={DashboardRoutes}
          />
        </Switch>
      </div>
    </Router>
  );
};
