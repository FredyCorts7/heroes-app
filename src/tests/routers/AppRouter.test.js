import React, { useContext } from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../routers/AppRouter';
import { AuthContext } from '../../auth/AuthContext';
import '@testing-library/jest-dom';

describe('Testing to <AppRouter />', () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      authenticated: false,
    },
  };

  test('should to show login if is not authenticated', () => {
    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('h1').text()).toBe('Login');
    expect(wrapper.find('button').exists()).toBeTruthy();
  });

  test('should to show MarvelScreen if is authenticated', () => {
    const contextValue = {
      dispatch: jest.fn(),
      user: {
        authenticated: true,
        name: 'Camila',
      },
    };

    const wrapper = mount(
      <AuthContext.Provider value={contextValue}>
        <AppRouter />
      </AuthContext.Provider>
    );

    // console.log(wrapper.html());
    expect(wrapper.find('.navbar').exists()).toBeTruthy();
  });
});
