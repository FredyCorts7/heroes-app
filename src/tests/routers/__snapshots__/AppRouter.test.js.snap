// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing to <AppRouter /> should to show login if is not authenticated 1`] = `
<div>
  <div
    className="container mt-5"
  >
    <h1>
      Login
    </h1>
    <hr />
    <button
      className="btn btn-info"
      onClick={[Function]}
    >
      Log in
    </button>
  </div>
</div>
`;
